### PR TITLE
[CPU] Select via Eltwise implementation

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - *Select reimplemented via Eltwise node*
 - *Select single layer test class refactored, dynamic shapes instances added*

### Tickets:
 - *104055*

### CODE REVIEW

**Comment by maxnick** [comment]
@a-sidorova , could you please review?

File: src/plugins/intel_cpu/src/cpu_types.cpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ const InferenceEngine::details::caseless_unordered_map<std::string, Type> type_t
         { "GridSample", Type::GridSample},
         { "OneHot", Type::OneHot},
         { "RegionYolo", Type::RegionYolo},
-        { "Select", Type::Select},
+        { "Select", Type::Eltwise},
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/src/cpu_types.cpp:146]: Minor:
I'd just to suggest to move `Select` to `Eltwise` [part](https://github.com/openvinotoolkit/openvino/blob/b793e1e8ae74595584a31f1dc6ccb7e11b92e588/src/plugins/intel_cpu/src/cpu_types.cpp#L28) of list

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp [hunk 2]
```diff
@@ -1,174 +1,206 @@
...
+        auto select = ngraph::builder::makeSelect(paramOuts, broadcast);
+
+        function = makeNgraphFunction(precision, parameters, select, "Eltwise");
+   }
+
+   void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        auto cond_tensor = utils::create_and_fill_tensor(funcInputs[0].get_element_type(), targetInputStaticShapes[0], 1, 0, 1);
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp:74]: To be honest I'm not sure that the tensor contains the both `true` and `false` here. I think there are only `false` values. By the way, I have locally checked values of tensor and I saw really only `false` values.

So I'd suggest to use the same [generation](https://github.com/openvinotoolkit/openvino/blob/bc663878eb5ecf2afb7a982ec85918285bc9c5b6/src/tests/functional/plugin/shared/src/snippets/select.cpp#L16) as the `Select` in `Snippets`. It generates different values. Also I'd suggest to generate different values for `then` and `else` inputs to see that `Select` correctly chooses values.

File: src/plugins/intel_cpu/src/cpu_types.cpp (hunk 1 above)
Comment (v-Golubev) [reply src/plugins/intel_cpu/src/cpu_types.cpp]: Done, thanks

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp (hunk 2 above)
Comment (v-Golubev) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp]: My bad. Thanks a lot for noticing this defect! Corrected

**Review by a-sidorova — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 3]
```diff
@@ -1660,6 +1666,13 @@ bool Eltwise::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op
                 return false;
             }
         }
+        if (const auto select = std::dynamic_pointer_cast<const ov::op::v1::Select>(op)) {
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/eltwise.cpp:1669]: mgraph RTTI is more preferred to use with ngraph types.
```suggestion
        if (const auto select = ov::as_type_ptr<const ov::op::v1::Select>(op)) {
```

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp [hunk 4]
```diff
@@ -1,174 +1,204 @@
...
+        inputs.insert({modelInputs[2].get_node_shared_ptr(), elseTensor});
+    }
+};
+
+TEST_P(SelectLayerCPUTest, CompareWithRefs) {
+    run();
+    CheckPluginRelatedResults(compiledModel, "Eltwise");
+}
+
+const std::vector<ElementType> precisions = {
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp:87]: Das it make sense to add `i8` precision as well?

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp (hunk 3 above)
Comment (v-Golubev) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp]: OK, I replaced dynamic cast with ngraph RTTI in eltwise.cpp for ngraph types

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp (hunk 4 above)
Comment (v-Golubev) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp:87]: Done, thanks

**Review by maxnick — APPROVED** [review]
Can be merged after internal validation

**Comment by v-Golubev** [comment]
@maxnick, internal validation passed. Details can be found in the ticket

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**